Source/View Controllers/UploadViewController.swift

```diff
@@ -230,22 +245,25 @@ class UploadViewController: NSViewController {
     }
 
     func updateCredentialsAvailable() {
-        guard credentialsAvailable != (
-            !jamfProServerLabel.stringValue.isEmpty
-            && (username != nil)
-            && !username.isEmpty
-            && (password != nil)
-            && !password.isEmpty) else { return }
-        credentialsAvailable = !credentialsAvailable
+        if let username = username, !username.isEmpty,
+           let password = password, !password.isEmpty,
+           !jamfProServerLabel.stringValue.isEmpty {
+            credentialsAvailable = true
+        } else {
+            credentialsAvailable = false
+        }
     }
 
     func updateReadForUpload() {
+        guard let payloadName = payloadName, !payloadName.isEmpty else {
+            readyForUpload = false
+            return
+        }
+
         guard readyForUpload != (
             credentialsVerified
             && credentialsAvailable
             && !organizationLabel.stringValue.isEmpty
-            && (payloadName != nil)
-            && !payloadName.isEmpty
             && (payloadIdentifier != nil)
```

Comment:
This line needs to be removed due to making `payloadIdentifier` not Optional.  The Xcode compiler caught this one.

Reply:
Removed the `nil` check but kept the `!payloadName.isEmpty` as maybe someone changes the initial value to be empty.

Reply:
Looking closer, I think the `!payloadName.isEmpty` check also needs to be put back in since that is a required field.  Sorry I didn't catch that one on the previous look at the code.

Reply:
This is catches in line 258. `guard let payloadName = payloadName, !payloadName.isEmpty else {`so it's already there. :)

Reply:
Ah, yes.  So it is.  Woohoo!